sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -87,31 +126,32 @@ namespace Azure { namespace Storage { namespace _internal {
 
   FileReader::~FileReader() { CloseHandle(static_cast<HANDLE>(m_handle)); }
 
-  FileWriter::FileWriter(const std::string& filename)
+  size_t FileReader::Read(uint8_t* buffer, size_t length, int64_t offset) const
   {
-    int sizeNeeded = MultiByteToWideChar(
-        CP_UTF8,
-        MB_ERR_INVALID_CHARS,
-        filename.data(),
-        static_cast<int>(filename.length()),
-        nullptr,
-        0);
-    if (sizeNeeded == 0)
+    length = std::min(length, static_cast<size_t>(std::max(0LL, m_fileSize - offset)));
+    if (length > std::numeric_limits<DWORD>::max())
     {
-      throw std::runtime_error("Invalid filename.");
+      throw std::runtime_error("Failed to read file.");
```

Comment:
Any way to add more detailed error messages, or at lease include the error code into the message?

Reply:
Yeah, we could do better. To be honest, I don't expect the operation will fail very often. I mean, you've opened the file successfully. Under which circumstances will the read operation fail? Maybe when you shrink the file while the library is trying to read it? But thinking from a different angle, it is kind of obvious if you edit a file that is being uploaded, then the upload is failed. A more detailed error message won't help a lot in this case maybe?